cmd/svcat/command/command.go

```diff
@@ -16,20 +16,38 @@ limitations under the License.
 
 package command
 
-import "github.com/spf13/cobra"
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
 
 // Command represents an svcat command.
 type Command interface {
+	// GetContext retrieves the command's context.
+	GetContext() *Context
+
 	// Validate and load the arguments passed to the svcat command.
 	Validate(args []string) error
 
 	// Run a validated svcat command.
 	Run() error
 }
 
+// NamespacedCommand represents a command that can be scoped to a namespace.
+type NamespacedCommand interface {
+	Command
+
+	// SetNamespace sets the effective namespace for the command.
+	SetNamespace(namespace string)
+}
+
 // PreRunE validates os args, and then saves them on the svcat command.
 func PreRunE(cmd Command) func(*cobra.Command, []string) error {
-	return func(_ *cobra.Command, args []string) error {
+	return func(c *cobra.Command, args []string) error {
```

Comment:
Ok, so if I'm reading this right, this worked in plugin mode because the plugin framework passes namespace in as an argument always? So when you run in standalone mode nothing was setting the namespace because it wasn't getting pulled from the context. 

Does this still work right if I'm running as a plugin?

Reply:
You've got it! As a plugin --namespace is ALWAYS set, so it wasn't a problem that we weren't falling back to the context's namespace when --namespace was unspecified.

After this fix, the plugin continues to work just as before, since it basically doesn't care about this logic at all.

Reply:
But wont it still go through this, see that the command is a namespaced command so it will enter the loop and overwrite the namespace on the command?

Reply:
Sorry, I was wrong. It _does_ go through the same logic but the end result is the same as it is today for kubectl plugin mode.

When in plugin mode --namespace will be set (always), so `DetermineNamespace` will return the value in --namespace, and then set that on the command.